---
title: "Supervised integration using SIGNAL"
author: "Yang Zhou"
date: "2024-06-06"
output: html_document
package: SIGNAL
vignette: |
  %\VignetteIndexEntry{SIGNAL: exploring group-specific technical variation patterns of single-cell data}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(collapse = FALSE)
```

# Introduction

SIGNAL integration requires at least one group variable, such as 'cell type,' 'developmental stage,' or 'tissue origin,' which is typically provided as metadata. Actually, SIGNAL only assumes homogeneity within subgroups under a given group variable.


If there is no additional group variable entirely, one can consider manually adding a ‘source’ variable which indicates whether the cells come from the same source. However, as mentioned before, if the dataset is highly heterogeneous, SIGNAL cannot effectively integrate the dataset. We introduce a metric, between-batch similarity, to quantify the homogeneity across batches within a dataset. A high between-batch similarity indicates homogeneity. In contrast, when between-batch similarity is low, indicating heterogeneity, SIGNAL integration with a manually added ‘source’ variable may not succeed. Next, we show this using the heterogeneous cell lines dataset.

```{r, message=FALSE, warning=FALSE}
library(SIGNAL)
library(irlba)
library(uwot)
library(ggpubr)
library(cowplot)
set.seed(2024)
```

# Load data matrix and metadata

We demonstrate SIGNAL integration on a commonly used single-cell RNA sequencing (scRNA-seq) [dataset](https://www.10xgenomics.com/datasets/) of cell lines. The highly variable genes (HVGs) are already selected and are used to perform integration. The dataset is highly heterogeneous across batches, as we can see the cell type compositions of the dataset.

```{r, message=FALSE, warning=FALSE}
X = readRDS("/home/server/zy/group_scripts/datasets_preparation/Jurkat_293t/X.rds")
meta = readRDS("/home/server/zy/group_scripts/datasets_preparation/Jurkat_293t/meta.rds")
table(meta[, c("CellType", "Batch")])
```

We also compute the between-batch similarity.

```{r, message=FALSE, warning=FALSE}
batch_similarity <- function(data, meta, b_factor) {
    scaled_data = SIGNAL:::scale_data(data)
    b_ns = unique(meta[, b_factor])
    b_means = sapply(b_ns, function(b) matrixStats::rowMedians(as.matrix(scaled_data[, which(meta[, b_factor] == b)] )))
    colnames(b_means) = b_ns
    b_means = scInt:::cosineNorm(b_means)
    all_b_means = rowMeans(b_means)
    mean(apply(b_means, 2, function(x) cor(x, all_b_means, method = 'pearson')))
}
print(round(batch_similarity(X, meta, 'Batch'), 2) )
```

# SIGNAL integration with a manually added 'source' variable

We manually add a 'source' variable for all cells in the data. SIGNAL cannot separate two cell types.

```{r, message=TRUE, warning=FALSE, fig.width=12, fig.height=5}
meta$Source = '0'
signal_emb = Run.gcPCA(X, meta, g_factor = "Source", b_factor = "Batch")
signal_umap = as.data.frame(umap(t(signal_emb)))
colnames(signal_umap) = c("UMAP1", "UMAP2")
signal_umap = cbind.data.frame(meta, signal_umap)
p1 = ggscatter(signal_umap, x = "UMAP1", y = "UMAP2", size = 0.1, color = "Batch", palette = "npg", legend = "right") + 
  guides(colour = guide_legend(override.aes = list(size = 2)))
p2 = ggscatter(signal_umap, x = "UMAP1", y = "UMAP2", size = 0.1, color = "CellType", palette = "npg", legend = "right") + 
  guides(colour = guide_legend(override.aes = list(size = 2)))
plot_grid(p1, p2, align = 'h', axis = "b")
```

We then simulate batch labels to construct a homogeneous dataset and integrate it using SIGNAL using 'source' as the group variable.

```{r, message=TRUE, warning=FALSE, fig.width=12, fig.height=5}
meta$NewLabel = as.character(meta$Batch)
meta$NewLabel[which(meta$NewLabel %in% c('Batch_1', 'Batch_2'))] = 'Simulated Batch_1'
meta$NewLabel[which(meta$NewLabel == 'Batch_3')] = 'Simulated Batch_2'
signal_emb = Run.gcPCA(X, meta, g_factor = "Source", b_factor = "NewLabel")
signal_umap = as.data.frame(umap(t(signal_emb)))
colnames(signal_umap) = c("UMAP1", "UMAP2")
signal_umap = cbind.data.frame(meta, signal_umap)
q1 = ggscatter(signal_umap, x = "UMAP1", y = "UMAP2", size = 0.1, color = "Batch", palette = "npg", legend = "right") + 
  guides(colour = guide_legend(override.aes = list(size = 2)))
q2 = ggscatter(signal_umap, x = "UMAP1", y = "UMAP2", size = 0.1, color = "CellType", palette = "npg", legend = "right") + 
  guides(colour = guide_legend(override.aes = list(size = 2)))
plot_grid(q1, q2, align = 'h', axis = "b")
```

# SIGNAL integration for highly homogeneous datasets without prior information

To integrate the heterogeneous single-cell data without prior information, we recommend using additional unsupervised integration methods such as our previously proposed scInt and clustering techniques like k-means to create coarse cell labels. Then SIGNAL can be operated as a supervised integration method.

<details>
  <summary>**Session Info**</summary>
  
```{r}
sessionInfo()
```

</details>
