<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="SIGNAL">
<title>Supervised integration using SIGNAL • SIGNAL</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Supervised integration using SIGNAL">
<meta property="og:description" content="SIGNAL">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SIGNAL</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Knowledge_transfer.html">Knowledge transfer from reference to query using SIGNAL</a>
    <a class="dropdown-item" href="../articles/Multiscale_analysis.html">Multi-scale analysis by multiple group variables</a>
    <a class="dropdown-item" href="../articles/Supervised_integration.html">Supervised integration using SIGNAL</a>
    <a class="dropdown-item" href="../articles/Unsupervised_integration.html">SIGNAL integration for unlabeled datasets</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/yzhou1999/SIGNAL/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Supervised integration using SIGNAL</h1>
                        <h4 data-toc-skip class="author">Yang Zhou</h4>
            
            <h4 data-toc-skip class="date">2024-06-06</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/yzhou1999/SIGNAL/blob/HEAD/vignettes/Supervised_integration.Rmd" class="external-link"><code>vignettes/Supervised_integration.Rmd</code></a></small>
      <div class="d-none name"><code>Supervised_integration.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>SIGNAL integration requires at least one group variable, such as
‘cell type,’ ‘developmental stage,’ or ‘tissue origin,’ which is
typically provided as metadata. Actually, SIGNAL only assumes
homogeneity within subgroups under a given group variable.</p>
<p>If there is no additional group variable entirely, one can consider
manually adding a ‘source’ variable which indicates whether the cells
come from the same source. However, as mentioned before, if the dataset
is highly heterogeneous, SIGNAL cannot effectively integrate the
dataset. We introduce a metric, between-batch similarity, to quantify
the homogeneity across batches within a dataset. A high between-batch
similarity indicates homogeneity. In contrast, when between-batch
similarity is low, indicating heterogeneity, SIGNAL integration with a
manually added ‘source’ variable may not succeed. Next, we show this
using the heterogeneous cell lines dataset.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/yzhou1999/SIGNAL" class="external-link">SIGNAL</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">irlba</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jlmelville/uwot" class="external-link">uwot</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">ggpubr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">2024</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-data-matrix-and-metadata">Load data matrix and metadata<a class="anchor" aria-label="anchor" href="#load-data-matrix-and-metadata"></a>
</h2>
<p>We demonstrate SIGNAL integration on a commonly used single-cell RNA
sequencing (scRNA-seq) <a href="https://www.10xgenomics.com/datasets/" class="external-link">dataset</a> of cell lines.
The highly variable genes (HVGs) are already selected and are used to
perform integration. The dataset is highly heterogeneous across batches,
as we can see the cell type compositions of the dataset.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">X</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"/home/server/zy/group_scripts/datasets_preparation/Jurkat_293t/X.rds"</span><span class="op">)</span></span>
<span><span class="va">meta</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"/home/server/zy/group_scripts/datasets_preparation/Jurkat_293t/meta.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">meta</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CellType"</span>, <span class="st">"Batch"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         Batch</span></span>
<span><span class="co">## CellType Batch_1 Batch_2 Batch_3</span></span>
<span><span class="co">##   293t      2885       0    1605</span></span>
<span><span class="co">##   jurkat       0    3258    1783</span></span></code></pre>
<p>We also compute the between-batch similarity.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">batch_similarity</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">data</span>, <span class="va">meta</span>, <span class="va">b_factor</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">scaled_data</span> <span class="op">=</span> <span class="fu">SIGNAL</span><span class="fu">:::</span><span class="fu"><a href="../reference/scale_data.html">scale_data</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span>
<span>    <span class="va">b_ns</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">meta</span><span class="op">[</span>, <span class="va">b_factor</span><span class="op">]</span><span class="op">)</span></span>
<span>    <span class="va">b_means</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">b_ns</span>, <span class="kw">function</span><span class="op">(</span><span class="va">b</span><span class="op">)</span> <span class="fu">matrixStats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/matrixStats/man/rowMedians.html" class="external-link">rowMedians</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">scaled_data</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">meta</span><span class="op">[</span>, <span class="va">b_factor</span><span class="op">]</span> <span class="op">==</span> <span class="va">b</span><span class="op">)</span><span class="op">]</span> <span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">b_means</span><span class="op">)</span> <span class="op">=</span> <span class="va">b_ns</span></span>
<span>    <span class="va">b_means</span> <span class="op">=</span> <span class="fu">scInt</span><span class="fu">:::</span><span class="fu">cosineNorm</span><span class="op">(</span><span class="va">b_means</span><span class="op">)</span></span>
<span>    <span class="va">all_b_means</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="va">b_means</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">b_means</span>, <span class="fl">2</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">all_b_means</span>, method <span class="op">=</span> <span class="st">'pearson'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu">batch_similarity</span><span class="op">(</span><span class="va">X</span>, <span class="va">meta</span>, <span class="st">'Batch'</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span> <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.51</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="signal-integration-with-a-manually-added-source-variable">SIGNAL integration with a manually added ‘source’ variable<a class="anchor" aria-label="anchor" href="#signal-integration-with-a-manually-added-source-variable"></a>
</h2>
<p>We manually add a ‘source’ variable for all cells in the data. SIGNAL
cannot separate two cell types.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta</span><span class="op">$</span><span class="va">Source</span> <span class="op">=</span> <span class="st">'0'</span></span>
<span><span class="va">signal_emb</span> <span class="op">=</span> <span class="fu"><a href="../reference/Run.gcPCA.html">Run.gcPCA</a></span><span class="op">(</span><span class="va">X</span>, <span class="va">meta</span>, g_factor <span class="op">=</span> <span class="st">"Source"</span>, b_factor <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Run gcPCA!</span></span></code></pre>
<pre><code><span><span class="co">## gcPCA done!</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">signal_umap</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://jlmelville.github.io/uwot/reference/umap.html" class="external-link">umap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">signal_emb</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">signal_umap</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMAP1"</span>, <span class="st">"UMAP2"</span><span class="op">)</span></span>
<span><span class="va">signal_umap</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind.data.frame</a></span><span class="op">(</span><span class="va">meta</span>, <span class="va">signal_umap</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html" class="external-link">ggscatter</a></span><span class="op">(</span><span class="va">signal_umap</span>, x <span class="op">=</span> <span class="st">"UMAP1"</span>, y <span class="op">=</span> <span class="st">"UMAP2"</span>, size <span class="op">=</span> <span class="fl">0.1</span>, color <span class="op">=</span> <span class="st">"Batch"</span>, palette <span class="op">=</span> <span class="st">"npg"</span>, legend <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span>override.aes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">=</span> <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html" class="external-link">ggscatter</a></span><span class="op">(</span><span class="va">signal_umap</span>, x <span class="op">=</span> <span class="st">"UMAP1"</span>, y <span class="op">=</span> <span class="st">"UMAP2"</span>, size <span class="op">=</span> <span class="fl">0.1</span>, color <span class="op">=</span> <span class="st">"CellType"</span>, palette <span class="op">=</span> <span class="st">"npg"</span>, legend <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span>override.aes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, align <span class="op">=</span> <span class="st">'h'</span>, axis <span class="op">=</span> <span class="st">"b"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Supervised_integration_files/figure-html/unnamed-chunk-4-1.png" width="1152"></p>
<p>We then simulate batch labels to construct a homogeneous dataset and
integrate it using SIGNAL using ‘source’ as the group variable.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta</span><span class="op">$</span><span class="va">NewLabel</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">meta</span><span class="op">$</span><span class="va">Batch</span><span class="op">)</span></span>
<span><span class="va">meta</span><span class="op">$</span><span class="va">NewLabel</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">meta</span><span class="op">$</span><span class="va">NewLabel</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Batch_1'</span>, <span class="st">'Batch_2'</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="op">=</span> <span class="st">'Simulated Batch_1'</span></span>
<span><span class="va">meta</span><span class="op">$</span><span class="va">NewLabel</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">meta</span><span class="op">$</span><span class="va">NewLabel</span> <span class="op">==</span> <span class="st">'Batch_3'</span><span class="op">)</span><span class="op">]</span> <span class="op">=</span> <span class="st">'Simulated Batch_2'</span></span>
<span><span class="va">signal_emb</span> <span class="op">=</span> <span class="fu"><a href="../reference/Run.gcPCA.html">Run.gcPCA</a></span><span class="op">(</span><span class="va">X</span>, <span class="va">meta</span>, g_factor <span class="op">=</span> <span class="st">"Source"</span>, b_factor <span class="op">=</span> <span class="st">"NewLabel"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Run gcPCA!</span></span></code></pre>
<pre><code><span><span class="co">## gcPCA done!</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">signal_umap</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://jlmelville.github.io/uwot/reference/umap.html" class="external-link">umap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">signal_emb</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">signal_umap</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMAP1"</span>, <span class="st">"UMAP2"</span><span class="op">)</span></span>
<span><span class="va">signal_umap</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind.data.frame</a></span><span class="op">(</span><span class="va">meta</span>, <span class="va">signal_umap</span><span class="op">)</span></span>
<span><span class="va">q1</span> <span class="op">=</span> <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html" class="external-link">ggscatter</a></span><span class="op">(</span><span class="va">signal_umap</span>, x <span class="op">=</span> <span class="st">"UMAP1"</span>, y <span class="op">=</span> <span class="st">"UMAP2"</span>, size <span class="op">=</span> <span class="fl">0.1</span>, color <span class="op">=</span> <span class="st">"Batch"</span>, palette <span class="op">=</span> <span class="st">"npg"</span>, legend <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span>override.aes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">q2</span> <span class="op">=</span> <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html" class="external-link">ggscatter</a></span><span class="op">(</span><span class="va">signal_umap</span>, x <span class="op">=</span> <span class="st">"UMAP1"</span>, y <span class="op">=</span> <span class="st">"UMAP2"</span>, size <span class="op">=</span> <span class="fl">0.1</span>, color <span class="op">=</span> <span class="st">"CellType"</span>, palette <span class="op">=</span> <span class="st">"npg"</span>, legend <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span>override.aes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">q1</span>, <span class="va">q2</span>, align <span class="op">=</span> <span class="st">'h'</span>, axis <span class="op">=</span> <span class="st">"b"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Supervised_integration_files/figure-html/unnamed-chunk-5-1.png" width="1152"></p>
</div>
<div class="section level2">
<h2 id="signal-integration-for-highly-homogeneous-datasets-without-prior-information">SIGNAL integration for highly homogeneous datasets without prior
information<a class="anchor" aria-label="anchor" href="#signal-integration-for-highly-homogeneous-datasets-without-prior-information"></a>
</h2>
<p>To integrate the heterogeneous single-cell data without prior
information, we recommend using additional unsupervised integration
methods such as our previously proposed scInt and clustering techniques
like k-means to create coarse cell labels. Then SIGNAL can be operated
as a supervised integration method.</p>
<details><summary><strong>Session Info</strong>
</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.3 (2023-03-15)</span></span>
<span><span class="co">## Platform: x86_64-conda-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.10</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS/LAPACK: /home/server/anaconda3/envs/zy/lib/libopenblasp-r0.3.21.so</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] cowplot_1.1.1  ggpubr_0.6.0   ggplot2_3.4.4  uwot_0.2.2     irlba_2.3.5.1 </span></span>
<span><span class="co">## [6] Matrix_1.5-4.1 SIGNAL_1.0.0  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] backports_1.4.1          systemfonts_1.0.6        RcppEigen_0.3.4.0.0     </span></span>
<span><span class="co">##   [4] plyr_1.8.9               igraph_1.3.1             lazyeval_0.2.2          </span></span>
<span><span class="co">##   [7] sp_2.1-3                 splines_4.2.3            BiocParallel_1.32.6     </span></span>
<span><span class="co">##  [10] listenv_0.9.1            scattermore_1.2          digest_0.6.35           </span></span>
<span><span class="co">##  [13] foreach_1.5.2            htmltools_0.5.8.1        fansi_1.0.6             </span></span>
<span><span class="co">##  [16] magrittr_2.0.3           memoise_2.0.1            tensor_1.5              </span></span>
<span><span class="co">##  [19] cluster_2.1.4            doParallel_1.0.17        ROCR_1.0-11             </span></span>
<span><span class="co">##  [22] bigassertr_0.1.6         bigstatsr_1.5.12         globals_0.16.3          </span></span>
<span><span class="co">##  [25] matrixStats_1.0.0        spatstat.sparse_3.0-2    pkgdown_2.0.7           </span></span>
<span><span class="co">##  [28] colorspace_2.1-0         ggrepel_0.9.5            textshaping_0.3.7       </span></span>
<span><span class="co">##  [31] xfun_0.43                dplyr_1.1.4              jsonlite_1.8.8          </span></span>
<span><span class="co">##  [34] spatstat.data_3.0-1      progressr_0.13.0         survival_3.5-5          </span></span>
<span><span class="co">##  [37] flock_0.7                zoo_1.8-12               iterators_1.0.14        </span></span>
<span><span class="co">##  [40] glue_1.7.0               polyclip_1.10-4          gtable_0.3.5            </span></span>
<span><span class="co">##  [43] leiden_0.4.3             car_3.1-2                future.apply_1.11.0     </span></span>
<span><span class="co">##  [46] BiocGenerics_0.44.0      rfunctions_0.1           abind_1.4-5             </span></span>
<span><span class="co">##  [49] scales_1.3.0             bigparallelr_0.3.2       spatstat.random_3.1-5   </span></span>
<span><span class="co">##  [52] rstatix_0.7.2            miniUI_0.1.1.1           Rcpp_1.0.12             </span></span>
<span><span class="co">##  [55] viridisLite_0.4.2        xtable_1.8-4             reticulate_1.26         </span></span>
<span><span class="co">##  [58] bit_4.0.5                mclust_6.0.0             stats4_4.2.3            </span></span>
<span><span class="co">##  [61] htmlwidgets_1.6.4        httr_1.4.7               RColorBrewer_1.1-3      </span></span>
<span><span class="co">##  [64] Seurat_4.3.0.1           ff_4.0.12                ica_1.0-3               </span></span>
<span><span class="co">##  [67] farver_2.1.1             pkgconfig_2.0.3          deldir_2.0-4            </span></span>
<span><span class="co">##  [70] sass_0.4.9               utf8_1.2.4               labeling_0.4.3          </span></span>
<span><span class="co">##  [73] tidyselect_1.2.1         rlang_1.1.3              reshape2_1.4.4          </span></span>
<span><span class="co">##  [76] later_1.3.2              munsell_0.5.1            tools_4.2.3             </span></span>
<span><span class="co">##  [79] cachem_1.0.8             cli_3.6.2                generics_0.1.3          </span></span>
<span><span class="co">##  [82] broom_1.0.5              ggridges_0.5.4           evaluate_0.23           </span></span>
<span><span class="co">##  [85] stringr_1.5.1            fastmap_1.1.1            goftest_1.2-3           </span></span>
<span><span class="co">##  [88] yaml_2.3.8               ragg_1.2.7               knitr_1.46              </span></span>
<span><span class="co">##  [91] fs_1.6.4                 fitdistrplus_1.1-11      purrr_1.0.2             </span></span>
<span><span class="co">##  [94] RANN_2.6.1               nlme_3.1-162             pbapply_1.7-2           </span></span>
<span><span class="co">##  [97] future_1.33.1            sparseMatrixStats_1.10.0 mime_0.12               </span></span>
<span><span class="co">## [100] compiler_4.2.3           rstudioapi_0.15.0        scInt_1.0               </span></span>
<span><span class="co">## [103] plotly_4.10.4            png_0.1-8                ggsignif_0.6.4          </span></span>
<span><span class="co">## [106] spatstat.utils_3.1-0     tibble_3.2.1             bslib_0.7.0             </span></span>
<span><span class="co">## [109] stringi_1.8.3            highr_0.10               ps_1.7.6                </span></span>
<span><span class="co">## [112] desc_1.4.3               RSpectra_0.16-1          lattice_0.21-8          </span></span>
<span><span class="co">## [115] ggsci_3.0.0              vctrs_0.6.5              pillar_1.9.0            </span></span>
<span><span class="co">## [118] lifecycle_1.0.4          spatstat.geom_3.2-1      lmtest_0.9-40           </span></span>
<span><span class="co">## [121] jquerylib_0.1.4          RcppAnnoy_0.0.22         BiocNeighbors_1.16.0    </span></span>
<span><span class="co">## [124] data.table_1.15.4        httpuv_1.6.15            patchwork_1.1.2         </span></span>
<span><span class="co">## [127] R6_2.5.1                 promises_1.3.0           KernSmooth_2.23-21      </span></span>
<span><span class="co">## [130] gridExtra_2.3            parallelly_1.37.1        codetools_0.2-19        </span></span>
<span><span class="co">## [133] MASS_7.3-60              withr_3.0.0              SeuratObject_4.1.3      </span></span>
<span><span class="co">## [136] sctransform_0.3.5        S4Vectors_0.36.2         parallel_4.2.3          </span></span>
<span><span class="co">## [139] grid_4.2.3               tidyr_1.3.1              rmio_0.4.0              </span></span>
<span><span class="co">## [142] rmarkdown_2.26           MatrixGenerics_1.10.0    carData_3.0-5           </span></span>
<span><span class="co">## [145] Rtsne_0.17               spatstat.explore_3.2-1   Biobase_2.58.0          </span></span>
<span><span class="co">## [148] shiny_1.8.1.1</span></span></code></pre>
</details>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Yang Zhou.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
